pkg/catalog/manager/crd.go

```diff
@@ -56,6 +76,10 @@ func (m *Manager) getTemplateMap(catalogName string, namespace string) (map[stri
 }
 
 func (m *Manager) updateTemplate(template *v3.CatalogTemplate, toUpdate v3.CatalogTemplate) error {
+	// check to see if the version has valid urls - do not process if none exist
+	if len(toUpdate.Spec.Versions) == 0 || toUpdate.Spec.Versions == nil {
+		return &InvalidChartError{"must have one or more valid urls", template.Name}
```

Comment:
The error doesn't convey what really happened, there isn't any data, not that it's invalid.

Reply:
removed as part of new implementation